cargo-dist/src/lib.rs

```diff
@@ -661,9 +664,23 @@ pub fn run_generate(dist: &DistGraph, args: &GenerateArgs) -> Result<()> {
 /// Run any necessary integrity checks for "primary" commands like build/plan
 ///
 /// (This is currently equivalent to `cargo dist generate --check`)
-pub fn check_integrity(dist: &DistGraph) -> Result<()> {
+pub fn check_integrity(cfg: &Config) -> Result<()> {
+    // We need to avoid overwriting any parts of configuration from CLI here,
+    // so construct a clean copy of config to run the check generate
+    let check_config = Config {
+        needs_coherent_announcement_tag: true,
```

Comment:
```suggestion
        needs_coherent_announcement_tag: false,
```

I was like "there was a weird thing about bypassing tags wasn't there..." and aha, here it is indeed. Setting this to false is basically "i work like init/generate/plan".